network/p2p/inspector/validation/control_message_validation_inspector.go

```diff
@@ -209,6 +209,23 @@ func (c *ControlMsgValidationInspector) Inspect(from peer.ID, rpc *pubsub.RPC) e
 		return nil
 	}
 
+	// check peer identity when running private network
+	// sanity check: rpc inspection should be disabled on public networks
+	if c.networkingType == network.PrivateNetwork {
+		err := c.checkSenderIdentity(from)
+		if err != nil {
+			c.notificationConsumer.OnInvalidControlMessageNotification(p2p.NewInvalidControlMessageNotification(from, p2pmsg.CtrlMsgGraft, err, 1, p2p.CtrlMsgNonClusterTopicType))
+			c.logger.
+				Error().
+				Err(err).
+				Str("peer_id", p2plogging.PeerId(from)).
+				Bool(logging.KeyNetworkingSecurity, true).
+				Msg("rpc received from unstaked peer")
+			c.metrics.OnInvalidControlMessageNotificationSent()
```

Comment:
Please also consider having a separate metric for this (`OnRpcRejectedFromUnknownSender`)

Reply:
https://github.com/onflow/flow-go/pull/5449/commits/c348b43c1ead93255fdc532d44a95e12894c25c5